resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientInvocation.java

```diff
@@ -161,7 +162,10 @@ public static <T> T extractResult(GenericType<T> responseType, Response response
          {
             if (response.getMediaType() == null)
             {
-               return null;
+               if (response.getStatus() == Response.Status.NO_CONTENT.getStatusCode()) {

```

Comment:
Thinking about this, I'm wondering now if we should throw the exception if the status is 200 or 203. Other statuses may rightfully not have a media type.